### PR TITLE
fix: only use array answers for populating previous answers

### PR DESCRIPTION
**Pull request by MasterKenth**
## Explain the changes you’ve made

Check if previous answers is array (i.e. not encrypted) before using for pre-populating new case.

## Explain why these changes are made

Some edge-cases have previous cases with decisions but with encrypted answers - pre-population then crashes for any new cases.

## How to test

Concrete example:

1. Checkout this branch
2. Have a case with decisions but replace the answers of the form with encrypted answers:

```json
"a1ce65a0-3d7c-11ec-801e-2db8270ee167": {
   "answers": {
    "encryptedAnswers": "abcdefg"
   },
   "currentPosition": {
    "currentMainStep": 4,
    "currentMainStepIndex": 3,
    "index": 3,
    "level": 0,
    "numberOfMainSteps": 9
   },
   "encryption": {
    "encryptionKeyId": "2fa16de5-6557-4f35-bb36-3a475fd9136d",
    "symmetricKeyName": "2fa16de5-6557-4f35-bb36-3a475fd9136d",
    "type": "password"
   }
  }
```

3. Login/refresh to try to create a new case and verify it is created as expected (with no pre-populated answers).

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-26b6gkr Hantera krypterade formulär så att lambdan inte krashar vid förpopulering](https://app.clickup.com/t/26b6gkr)

File: libs/formAnswers.js [hunk 1]
```diff
@@ -167,7 +167,9 @@ export function populateFormWithPreviousCaseAnswers(
   Object.keys(forms).forEach(formId => {
     const caseFormTemplate = formTemplates?.[formId] || {};
 
-    const previousAnswers = previousForms?.[formId]?.answers || [];
+    const previousCaseAnswers = previousForms?.[formId]?.answers;
+    const previousAreEncrypted = !Array.isArray(previousCaseAnswers);
```
Comment (dannil76) [inline libs/formAnswers.js:171]: Tycker det är bra om vi håller oss till formatet med isVariableName. Missade själv att det stod Are i mitten av ordet.

isPreviousAnswersEncrypted blir väl bra?

**Review by dannil76 — CHANGES_REQUESTED** [review]
Linting gnäller

File: libs/formAnswers.js (hunk 1 above)
Comment (MasterKenth) [reply libs/formAnswers.js:171]: Kan köra på `isPreviousEncrypted` annars blir det kass engelska eller för verbost.

File: libs/formAnswers.d.ts [hunk 2]
```diff
@@ -12,9 +12,9 @@ export function formatAnswer(
...
 export function mergeAnswers(previousAnswers: any, newAnswers: any): any[];
-export function populateFormWithPreviousCaseAnswers(
+export function populateFormWithPreviousCaseAnswers({
   forms: any,
   applicants: any,
   formTemplates: any,
-  previousForms: any
-): Record<string, never>;
+  previousForms: any,
+}): Record<string, never>;
```
Comment (kajm) [inline libs/formAnswers.d.ts:20]: Can these be typed?

**Comment by helsingborgsstad** [comment]
Task linked: [CU-3n5jccc Skapa logik som kontrollerar personnnummer mot en lista i ssm-params](https://app.clickup.com/t/3n5jccc)
